Packs/SilentPush/Integrations/SilentPush/SilentPush.py

```diff
@@ -623,68 +802,129 @@ def density_lookup(self, qtype: str, query: str, **kwargs) -> Dict[str, Any]:
 
         return response
 
+    def screenshot_url(self, url: str) -> Dict[str, Any]:
+        """
+        Generate a screenshot for a given URL and store it in the vault using GET request.
+
+        Args:
+            url (str): The URL to capture a screenshot of
+
+        Returns:
+            Dict[str, Any]: Response containing screenshot information and vault details
+        """
+        endpoint = "explore/tools/screenshotondemand"
+        params = {"url": url}
+
+        try:
+
+            response = self._http_request(
+                method="GET",
+                url_suffix=endpoint,
+                params=params
+            )
+
+            if response.get("error"):
+                raise DemistoException(f"Failed to get screenshot: {response['error']}")
+
+            screenshot_data = response.get("response", {}).get("screenshot", {})
+            if not screenshot_data:
+                raise DemistoException("No screenshot data returned from API")
+
+            screenshot_url = screenshot_data.get("message")
+            if not screenshot_url:
+                raise DemistoException("No screenshot URL returned")
+
+
+            image_response = requests.get(screenshot_url, verify=self.verify)
+            if image_response.status_code != 200:
+                raise DemistoException(f"Failed to download screenshot image: HTTP {image_response.status_code}")
+
+
+            filename = f"{url.split('://')[1].split('/')[0]}_screenshot.jpg"
+            return {
+                "status_code": screenshot_data.get("response", 200),
+                "screenshot_url": screenshot_url,
+                "vault_info": fileResult(filename, image_response.content),
+                "filename": filename
+            }
+
+        except Exception as e:
+            raise DemistoException(f"Error capturing screenshot: {str(e)}")
 
 def test_module(client: Client) -> str:
     try:
-        client.list_domain_information('silentpush.com')
+        resp = client.search_domains()
+        if resp.get("status_code") != 200:
```

Comment:
Ensure response is a dictionary and check status code


```suggestion
        if not isinstance(resp, dict) or resp.get("status_code") != 200:

```